### PR TITLE
`consul-k8s` CLI Envoy Debugging: Read Command

### PR DESCRIPTION
**Pull request by t-eckert**
## Changes proposed in this PR

- Add a command `consul-k8s proxy read podname` which prints a human-readable summary of the configuration for the Envoy sidecar in the target Pod.
- Add a struct which parses in the raw config from Envoy and isolates the relevant information for display.
- Add unit tests for the command and the config struct.

## Demo

https://user-images.githubusercontent.com/29112081/174871901-d3457c96-9097-443e-83ab-c85fde1a63c3.mov

## Last few tasks I'm finishing up as you review

- [x] Properly parsing filter and filter chain match config and displaying both in a table.
- [x] Properly parsing secrets config and displaying the secrets in a table.

## Not in scope for this PR

- Changelog entry
- Acceptance tests for the command
- Filtering by FQDN or Port

These will come in follow-up PRs.

## How I've tested this PR

- Unit tests.
- Manually exercising the CLI.

## How I expect reviewers to test this PR

- Unit tests can be run with `go test ./...` from the `cli` directory.
- The new command can be manually tested with

## Checklist
- [x] Tests added

### CODE REVIEW

File: cli/cmd/proxy/read/config_test.go [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+	var envoyConfig EnvoyConfig
+	err = json.Unmarshal(raw, &envoyConfig)
+	require.NoError(t, err)
+
+	actual := envoyConfig.JSON()
+
+	require.Equal(t, expected, actual)
+}
+
+func TestFetchConfig(t *testing.T) {
```
Comment (jmurret) [inline cli/cmd/proxy/read/config_test.go:50]: 👏

Comment (ndhanushkodi) [reply cli/cmd/proxy/read/config_test.go:50]: +1!!

**Review by ndhanushkodi — APPROVED** [review]
This looks amazing Thomas!! +1 on how clean it is too!

File: cli/cmd/proxy/read/config.go [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+}
+
+// JSON returns the original JSON Envoy config dump data which was used to create
+// the Config object.
+func (c *EnvoyConfig) JSON() []byte {
+	return c.rawCfg
+}
+
+// UnmarshalJSON unmarshals the raw config dump bytes into EnvoyConfig.
+func (c *EnvoyConfig) UnmarshalJSON(b []byte) error {
```
[the remark this answers is not in the document]
Comment (t-eckert) [reply cli/cmd/proxy/read/config.go:107]: Totally!

File: cli/cmd/proxy/read/config_test.go (hunk 1 above)
Comment (t-eckert) [reply cli/cmd/proxy/read/config_test.go:50]: Thank you!

**Comment by t-eckert** [comment]
That makes Envoy Debugging feature complete for the beta.

**Pull request merged**